pkg/kubectl/history.go

```diff
@@ -127,3 +130,71 @@ func getChangeCause(obj runtime.Object) string {
 	}
 	return accessor.GetAnnotations()[ChangeCauseAnnotation]
 }
+
+type DaemonSetHistoryViewer struct {
+	c clientset.Interface
+}
+
+// ViewHistory returns a revision-to-replicaset map as the revision history of a deployment
+func (h *DaemonSetHistoryViewer) ViewHistory(namespace, name string, revision int64) (string, error) {
+	daemon, err := h.c.Extensions().DaemonSets(namespace).Get(name)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+	}
+	podTemplates, err := daemonutil.GetAllPodTemplates(daemon, h.c)
```

Comment:
I would prefer to see using the pod template client directly instead of creating wrappers around it. We had been doing this in OpenShift deployments for some time and it was really painful to maintain.